### ISSUE TITLE
Address family not supported by protocol

### ISSUE THREAD

**Issue by CrossLee1** [rg3/youtube-dl#11081]
- [x] I've **verified** and **I assure** that I'm running youtube-dl **2016.10.31**
- [x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

- [x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [x] Question
- [ ] Other

./youtube-dl -v https://www.youtube.com/watch?v=OapMBWNgTzU
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'https://www.youtube.com/watch?v=OapMBWNgTzU']
[debug] Encodings: locale ISO-8859-1, fs ISO-8859-1, out ISO-8859-1, pref ISO-8859-1
[debug] youtube-dl version 2016.10.31
[debug] Python version 2.7.3 - Linux-2.6.32_1-15-0-0-x86_64-with-redhat-4.3-Final
[debug] exe versions: none
[debug] Proxy map: {}
[youtube] OapMBWNgTzU: Downloading webpage
**ERROR: Unable to download webpage: <urlopen error [Errno 97] Address family not supported by protocol> (caused by URLError(error(97, 'Address family not supported by protocol'),))**
  File "./youtube-dl/youtube_dl/extractor/common.py", line 403, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "./youtube-dl/youtube_dl/YoutubeDL.py", line 2001, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/home/users/lifu//lib/python2.7/urllib2.py", line 400, in open
    response = self._open(req, data)
  File "/home/users/lifu//lib/python2.7/urllib2.py", line 418, in _open
    '_open', req)
  File "/home/users/lifu//lib/python2.7/urllib2.py", line 378, in _call_chain
    result = func(*args)
  File "./youtube-dl/youtube_dl/utils.py", line 1036, in https_open
    req, **kwargs)
  File "/home/users/lifu//lib/python2.7/urllib2.py", line 1177, in do_open
    raise URLError(err)

**Comment by yan12125** [comment]
Could you try:
```
./youtube-dl --force-ipv4 -v https://www.youtube.com/watch?v=OapMBWNgTzU
```

**Comment by CrossLee1** [comment]
@yan12125 
Thanks, I tried following your suggestion, `./youtube-dl --force-ipv4 -v https://www.youtube.com/watch?v=OapMBWNgTzU`, but still got the problem.

[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'--force-ipv4', u'-v', u'https://www.youtube.com/watch?v=OapMBWNgTzU']
[debug] Encodings: locale ISO-8859-1, fs ISO-8859-1, out ISO-8859-1, pref ISO-8859-1
[debug] youtube-dl version 2016.10.31
[debug] Python version 2.7.3 - Linux-2.6.32_1-15-0-0-x86_64-with-redhat-4.3-Final
[debug] exe versions: none
[debug] Proxy map: {}
[youtube] OapMBWNgTzU: Downloading webpage
**ERROR: Unable to download webpage: <urlopen error [Errno 97] Address family not supported by protocol> (caused by URLError(error(97, 'Address family not supported by protocol'),))**
  File "./youtube-dl/youtube_dl/extractor/common.py", line 403, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "./youtube-dl/youtube_dl/YoutubeDL.py", line 2001, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/home/users/lifu//lib/python2.7/urllib2.py", line 400, in open
    response = self._open(req, data)
  File "/home/users/lifu//lib/python2.7/urllib2.py", line 418, in _open
    '_open', req)
  File "/home/users/lifu//lib/python2.7/urllib2.py", line 378, in _call_chain
    result = func(*args)
  File "./youtube-dl/youtube_dl/utils.py", line 1036, in https_open
    req, **kwargs)
  File "/home/users/lifu//lib/python2.7/urllib2.py", line 1177, in do_open
    raise URLError(err)

**Comment by yan12125** [comment]
Need more information:
```
strace -f ./youtube-dl --force-ipv4 -v https://www.youtube.com/watch?v=OapMBWNgTzU
```
This command prints lots of additional information. Could you save all outputs to a file and upload it somewhere?

**Comment by CrossLee1** [comment]
@yan12125 
Please refer to [this](https://drive.google.com/file/d/0B_JxpHb2vcJTeGp4Y1JLR1EzdTA/view?usp=sharing) for the outputs information. 
Thanks for your help~~

**Comment by yan12125** [comment]
Seems your DNS server is contaminated. Can you watch this video in browsers? Do you use any VPN or proxy tools?

**Comment by CrossLee1** [comment]
I can watch this video in browser on my Windows 7 and uses a proxy tool.
But on my server, I did not use any VPN or proxy.

**Comment by yan12125** [comment]
Let me list what I find from your log and possible solutions:

* Your server "cq01-rdqa-dev003.cq01.baidu.com" is provided by Baidu, a China company. I guess the server is in China. AFAIK devices in China can't access YouTube because of GFW, the Great Firewall of China
* On your server, www.youtube.com is resolved to 93.46.8.89. According to [Wikipedia](https://zh.wikipedia.org/wiki/%E5%9F%9F%E5%90%8D%E6%9C%8D%E5%8A%A1%E5%99%A8%E7%BC%93%E5%AD%98%E6%B1%A1%E6%9F%93) (Chinese article), this IP is used by GFW to prevent people in China from accessing YouTube and other blocked sites

Above all, you need a proxy on your server, too. FYI, youtube-dl's ```--proxy``` option is useful.

**Comment by yan12125** [comment]
As there's no response for a long time, I guess the problem is already solved. Feel free to leave comments or open a new issue if there are still problems.
